packages/mdc-data-table/README.md

```diff
@@ -1,4 +1,258 @@
+<!--docs:
+title: "Data Tables"
+layout: detail
+section: components
+excerpt: "Material Design-styled tables."
+iconId: data_table
+path: /catalog/data-tables/
+-->
+
 # Data Table
 
-The [Data Table component](https://material.io/go/design-data-tables) is yet to be completed, please follow the [tracking issue](https://github.com/material-components/material-components-web/issues/57) for more information.
+Data tables display sets of data.
+
+## Design & API Documentation
+
+<ul class="icon-list">
+  <li class="icon-list-item icon-list-item--spec">
+    <a href="https://material.io/go/design-data-tables">Material Design guidelines: Data tables</a>
+  </li>
+</ul>
+
+## Installation
+
+```
+npm install @material/data-table
+```
+
+### Styles
+
+```scss
+@import "@material/data-table/mdc-data-table";
+```
+
+> *NOTE*: Styles for any components you intend to include within data-table (e.g. Checkboxes, etc.) must also be
+> imported.
+
+### JavaScript Instantiation
+
+```js
+import {MDCDataTable} from '@material/data-table';
+const dataTable = new MDCDataTable(document.querySelector('.mdc-data-table'));
+```
+
+> See [Importing the JS component](../../docs/importing-js.md) for more information on how to import JavaScript.
+
+> Instantiating `MDCDataTable` component is only required to add interactions for example, row selection.
+
+MDC Data Table component auto instantiates `MDCCheckbox` for header row checkbox and all row checkboxes. Make sure to set required class names to instantiate checkbox component. We suggest to use `layout` API when rows are added or removed from data table to register new checkbox components.
```

Comment:
I think add a header here, like: `#### Usage with Checkboxes`

Reply:
I did not want to separate this paragraph from above snippet. It is very closely related to JS instantiation section which is only applicable for row selection as of now.